app/controllers/chargeback_controller.rb

```diff
@@ -597,23 +708,44 @@ def cb_rate_set_form_vars
   def cb_rate_get_form_vars
     @sb[:rate] = @edit[:rate]
     @edit[:new][:description] = params[:description] if params[:description]
-    @edit[:new][:details].each_with_index do |_detail, i|
-      @edit[:new][:details][i][:rate] = params["rate_#{i}".to_sym] if params["rate_#{i}".to_sym]
-      @edit[:new][:details][i][:per_time] = params["per_time_#{i}".to_sym] if params["per_time_#{i}".to_sym]
-      @edit[:new][:details][i][:per_unit] = params["per_unit_#{i}".to_sym] if params["per_unit_#{i}".to_sym]
+    @edit[:new][:details].each_with_index do |detail, detail_index|
+      %i{per_time per_unit}.each do |measure|
+        key = "#{measure}_#{detail_index}".to_sym
+        detail[measure] = params[key] if params[key]
+      end
       # Add currencies to chargeback_controller.rb
-      @edit[:new][:details][i][:currency] = params[:currency] if params[:currency]
+      detail[:currency] = params[:currency] if params[:currency]
+
+      # Save tiers into @edit
+      (0..@sb[:num_tiers][detail_index].to_i - 1).each do |tier_index|
+        tier = @edit[:new][:tiers][detail_index][tier_index] || {}
+        %i{fixed_rate variable_rate start end}.each do |field|
+          key = "#{field}_#{detail_index}_#{tier_index}".to_sym
+          tier[field] = params[key] if params[key]
+        end
+      end
     end
   end
 
   def cb_rate_set_record_vars
-    @edit[:new][:details].each_with_index do |_rate, i|
-      @sb[:rate_details][i].rate               = @edit[:new][:details][i][:rate]
-      @sb[:rate_details][i].per_time           = @edit[:new][:details][i][:per_time]
-      @sb[:rate_details][i].per_unit           = @edit[:new][:details][i][:per_unit]
+    @edit[:new][:details].each_with_index do |_detail, detail_index|
+      @sb[:rate_details][detail_index].per_time           = _detail[:per_time]
+      @sb[:rate_details][detail_index].per_unit           = _detail[:per_unit]
       # C: Record the currency selected in the edit view, in my chargeback_rate_details table
-      @sb[:rate_details][i].chargeback_rate_detail_currency_id = @edit[:new][:details][i][:currency]
-      @sb[:rate_details][i].chargeback_rate_id = @sb[:rate].id
+      @sb[:rate_details][detail_index].chargeback_rate_detail_currency_id = @edit[:new][:details][detail_index][:currency]
+      @sb[:rate_details][detail_index].chargeback_rate_id = @sb[:rate].id
+      # Save tiers into @sb
+      @edit[:new][:tiers][detail_index].each_with_index do |_tier, tier_index|
+        @sb[:tiers][detail_index][tier_index] =
```

Comment:
@rixhack @gtanzillo 
```ruby
edit[:new][:tiers][detail_index].each_with_index do |_tier, tier_index|
        _tier[:end] = ChargebackTier.to_float(_tier[:end]).to_f if _tier[:end].kind_of?(String)
        ...
```

I think we need to do something this, it solves scenario when you are trying to write "Infinity" into any `end` input.
otherwise it display validations errors